spec/requests/api/v1/timelines/public_spec.rb

```diff
@@ -95,25 +95,28 @@
     end
 
     context 'when the instance does not allow public preview' do
+      let(:expected_statuses) { [local_status, remote_status, media_status] }
+
       before do
         Form::AdminSettings.new(timeline_preview: false).save
       end
 
       context 'with an authenticated user' do
-        let(:expected_statuses) { [local_status, remote_status, media_status] }
-
         it_behaves_like 'a successful request to the public timeline'
       end
 
-      context 'with an unauthenticated user' do
-        let(:headers) { {} }
+      context 'with an authenticated user but using the wrong scope' do
+        it_behaves_like 'forbidden for wrong scope', 'follow'
+      end
 
-        it 'returns http unprocessable entity' do
-          subject
+      context 'with an authenticated application' do
+        let(:client_app) { Fabricate(:application) }
+        let(:token) { Fabricate(:accessible_access_token, application: client_app, scopes: scopes) }
 
-          expect(response).to have_http_status(422)
-        end
+        it_behaves_like 'a successful request to the public timeline'
```

Comment:
This is only correct if we _do_ want to allow App token access, otherwise this test case should read:

```
it_behaves_like 'unprocessable entity'
```

Since we don't have an explicit assertion in `require_user!` to assert that the access token has a `resource_owner_id` (i.e., User token vs App token).

Additionally, I'm not sure if `422` is the correct status code to be returning in: https://github.com/mastodon/mastodon/blob/1d3ecd3fbaf15450890c1ece7955846fba82a58d/app/controllers/api/base_controller.rb#63

It feels like that should be a `403` instead, since the request is present an access token, just not the type of token we expect.

Reply:
I think we indeed want to require a user.

Reply:
Okay, I can update this PR hopefully later today to do that